Xamarin.Forms.Core/TouchGestureRecognizer.cs

```diff
@@ -0,0 +1,223 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+
+namespace Xamarin.Forms
+{
+	public class TouchGestureRecognizer : GestureRecognizer
```

Comment:
I understand the need of using this as a base class for all your Gestures, but I still think all the Gestures should inherit from GestureRecognizer directly. this thus could be turned into an (internal) interface, and a couple of static method in a helper class

Reply:
I would like provide to users to create their own gesture recognizer based on TouchGestureRecognizer.
Therefore, making this class internal will be wrong.
I can also add all this functionality to GestureRecognizer if it is not superfluous.

Reply:
done